AsyncDisplayKit/TextKit/ASTextKitRenderer.mm

```diff
@@ -144,9 +162,18 @@ - (void)_calculateSize
   if (isinf(_constrainedSize.width) == NO && [_attributes.pointSizeScaleFactors count] > 0) {
     _currentScaleFactor = [[self fontSizeAdjuster] scaleFactor];
   }
-  
-  __block NSTextStorage *scaledTextStorage = nil;
+
+  // If we do not scale, do exclusion, or do custom truncation, we should just use TextKit for a fast-path.
   BOOL isScaled = [self isScaled];
+  if (isScaled == NO && self.usesCustomTruncation == NO && self.usesExclusionPaths == NO) {
+    CGRect rect = [_attributes.attributedString boundingRectWithSize:_constrainedSize options:NSStringDrawingUsesLineFragmentOrigin | NSStringDrawingTruncatesLastVisibleLine context:self.stringDrawingContext];
```

Comment:
Is it worth checking if attributedString is nil?  I think recent LLVM makes it safe to use the struct returned by messaging nil, but historically this was not safe, and I'm not totally sure.